types/mixpanel-browser/index.d.ts

```diff
@@ -159,8 +163,8 @@ export interface Mixpanel {
     opt_in_tracking(options?: Partial<InTrackingOptions>): void;
     opt_out_tracking(options?: Partial<OutTrackingOptions>): void;
     push(item: PushItem): void;
-    register(props: Dict, days?: number): void;
-    register_once(props: Dict, default_value?: any, days?: number): void;
+    register(props: Dict, daysOrOptions?: number | RegisterOptions): void;
```

Comment:
These lines seem related to https://github.com/mixpanel/mixpanel-js/compare/v2.39.0...v2.40.0#diff-7ee347fb24b1a0fafa61ea847381bcbaf393aa64f90e2c28aa99c56ae08b2ce8R7411-R7425.

```js
     *     // register only for the current pageload
     *     mixpanel.register({'Name': 'Pat'}, {persistent: false});
     *
     * @param {Object} properties An associative array of properties to store about the user
     * @param {Number|Object} [days_or_options] Options object or number of days since the user's last visit to store the super properties (only valid for persisted props)
     * @param {boolean} [days_or_options.days] - number of days since the user's last visit to store the super properties (only valid for persisted props)
     * @param {boolean} [days_or_options.persistent=true] - whether to put in persistent storage (cookie/localStorage)
     */
    MixpanelLib.prototype.register = function(props, days_or_options) {
        var options = options_for_register(days_or_options);
        if (options['persistent']) {
            this['persistence'].register(props, options['days']);
        } else {
            _.extend(this.unpersisted_superprops, props);
        }
    };
```

How about following the convention?

```suggestion
    register(props: Dict, days_or_options?: number | RegisterOptions): void;
```